WalletWasabi.Packager/MacSignTools.cs

```diff
@@ -327,85 +328,22 @@ private static Process WaitProcessToFinish(Process? process, string processName)
+	private static void Notarize(string appleId, string filePath)
 	{
-		string? uploadId = null;
-
 		Console.WriteLine("Start notarizing, uploading file.");
 
-		using (var process = Process.Start(new ProcessStartInfo
+		// -p WasabiNotarize = Saved the credentials in the keychain profile which keeps the password safe on the local machine. Name of the profile is "WasabiNotarize".
+		using var process = Process.Start(new ProcessStartInfo
 		{
 			FileName = "xcrun",
-			Arguments = $"altool --notarize-app -t osx -f \"{filePath}\" --primary-bundle-id \"{bundleIdentifier}\" -u \"{appleId}\" -p \"{password}\" --output-format xml",
+			Arguments = $"notarytool submit --wait --apple-id \"{appleId}\" -p \"WasabiNotarize\" \"{filePath}\" ",
 			RedirectStandardOutput = true,
-		}))
-		{
-			var nonNullProcess = WaitProcessToFinish(process, "xcrum");
-			string result = nonNullProcess.StandardOutput.ReadToEnd();
-
-			if (result.Contains("The software asset has already been uploaded. The upload ID is"))
-			{
-				// Example: The software asset has already been uploaded. The upload ID is 7689dc08-d6c8-4783-8d28-33e575f5c967
-				uploadId = result.Split('"').First(line => line.Contains("The software asset has already been uploaded.")).Split("The upload ID is")[^1].Trim();
-			}
-			else if (result.Contains("No errors uploading"))
-			{
-				// Example: <key>RequestUUID</key>\n\t\t<string>2a2a164f-2ae7-4293-8357-5d5a5cdd580a</string>
-
-				var lines = result.Split('\n');
-
-				for (int i = 0; i < lines.Length; i++)
-				{
-					string line = lines[i].Trim();
-					if (!line.TrimStart().StartsWith("<key>", StringComparison.InvariantCultureIgnoreCase))
-					{
-						continue;
-					}
-
-					if (line.Contains("<key>RequestUUID</key>", StringComparison.InvariantCulture))
-					{
-						uploadId = lines[i + 1].Trim().Replace("<string>", "").Replace("</string>", "");
-					}
-				}
-			}
-		}
-
-		if (uploadId is null)
-		{
-			throw new InvalidOperationException("Cannot get uploadId. Notarization failed.");
-		}
+		});
 
-		Stopwatch sw = new();
-		sw.Start();
-		while (true) // Wait for the notarization.
-		{
-			Console.WriteLine($"Checking notarization status. Elapsed time: {sw.Elapsed}");
-			using var process = Process.Start(new ProcessStartInfo
-			{
-				FileName = "xcrun",
-				Arguments = $"altool --notarization-info \"{uploadId}\" -u \"{appleId}\" -p \"{password}\"",
-				RedirectStandardError = true,
-				RedirectStandardOutput = true,
-			});
-			var nonNullProcess = WaitProcessToFinish(process, "xcrum");
-			string result = $"{nonNullProcess.StandardError.ReadToEnd()} {nonNullProcess.StandardOutput.ReadToEnd()}";
-			if (result.Contains("Status Message: Package Approved"))
-			{
-				break;
-			}
-			if (result.Contains("Status: in progress"))
-			{
-				Thread.Sleep(4000);
-				continue;
-			}
-			if (result.Contains("Could not find the RequestUUID"))
-			{
-				Thread.Sleep(4000);
-				continue;
-			}
```

Comment:
No need to periodically send a request to Apple about the notarization's status.

Thanks to the `--wait` option.